const User = require('../models/User')

module.exports.list = async (req, res) => {

    try {
        console.log("ðŸš€ ~ list here: ")

        const users = await User.find()
        console.log("ðŸš€ ~ users", users)
       
        res.send({success: true, users})
    } catch (error) {
        
        console.log("ðŸš€ ~ Error in list users", error.message)
        res.send({success: false, error: error.message})

    }
}

module.exports.register = async (req, res) => {
    try {
        
        console.log("ðŸš€ ~ register here: ", req.body)

        const {email, username} = req.body;

        if (!email || !username) {
            res.send({success: false, error: 'validation failed'})
            return
        }

        const userCreated = await User.create(req.body)
        console.log("ðŸš€ ~ userCreated", userCreated)

        res.send({success: true})

    } catch (error) {
    
        console.log("ðŸš€ ~ Error in register", error.message)

        res.send({success: false, error: error.message})
        
    }
}

module.exports.findById = async (req, res) => {
    try {
        console.log("req params", req.params)
        if (!req.params.userId) {
            throw new Error("No User found for this id")
        }
        const user = await User.findById(req.params.userId)
        res.send({success: true, user})
    } catch (error) {
        console.log("ðŸš€ ~ Error while finding User", error.message)

        res.send({success: false, error: error.message})
    }
}